SNIClient.py

```diff
@@ -649,7 +652,11 @@ async def game_watcher(ctx: SNIContext) -> None:
 
         perf_counter = time.perf_counter()
 
-        await ctx.client_handler.game_watcher(ctx)
+        try:
+            await ctx.client_handler.game_watcher(ctx)
+        except Exception as ex:
+            ctx.gui_error("Game Exception", ex)
+            await snes_disconnect(ctx)
```

Comment:
Any time an exception happens, I think the full stack trace should go in the text file log (in addition to the shorter message displayed to the user).

This can be done by using the default logger:
```python
            text_file_logger = logging.getLogger()
            text_file_logger.exception(ex)
```

Reply:
I was actually under the impression that ctx.gui_error actually logged the error, since it would make sense to do that while showing the visual.